### ISSUE TITLE
Issue running docusaurus v2 on github pages behind a proxy

### ISSUE THREAD

**Issue by jmcshane** [facebook/docusaurus#2719]
## 🐛 Bug Report

When running behind an nginx proxy, the github pages site responds with a redirect to github pages rather than serving the content through the proxy.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes
## To Reproduce

1. Nginx config:

        location / {
            set $doc_site_url "${doc_site_host}/pages/jmcshane/docs/${request_uri}";
            # redirect .html to raw
            if ($request_uri ~ ^(?:/(docs/.*)\.html|/(\w+).html)$) {
                return 302 https://$http_host/$1$2;
            }
            proxy_pass $doc_site_url;
        }

1. Access nginx url
1. Get redirect to github enterprise site:

$ curl http://localhost:8080/publiccloud -vv
*   Trying ::1...
* TCP_NODELAY set
* Connection failed
* connect to ::1 port 8080 failed: Connection refused
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /page HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 301 Moved Permanently
< Server: nginx/1.17.10
< Date: Mon, 04 May 2020 18:08:51 GMT
< Content-Type: text/html
< Content-Length: 162
< Connection: keep-alive
< Location: http://github.<myorg>.com/pages/jmcshan2/docs/page
< Access-Control-Allow-Origin: *
< Expires: Mon, 04 May 2020 18:18:57 GMT
< Cache-Control: max-age=600

## Expected behavior

Respond with the page content

## Actual Behavior

Received a 301, see above

## Your Environment

- Docusaurus version used: v2-alpha.50
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0): Chrome: 81.0.4044.129 Nodejs 14.0
- Operating system and version (desktop or mobile): Desktop

## Reproducible Demo

I apologize, I cannot share the entire repo. I can link the main sections of the site config below:

```
module.exports = {
  title: 'My v2 site',
  tagline: 'Test V2',
  url: 'https://mydocs.myorg.com', // this is what needs to get resolved from nginx
  baseUrl: '/',
  favicon: 'img/favicon.png',
  organizationName: 'jmcshane', // Usually your GitHub org/user name.
  projectName: 'docs', // Usually your repo name.
}
```

**Comment by jmcshane** [comment]
This appears to be caused by the jekyll structure that gets generated with v2 vs v1. When generating the pages in v1, markdown documents get created as `/docs/mypage.html`. When generating pages in v2, the same pages are built as `/docs/mypage/index.html`. When requests go to the Jekyll server to `/docs/mypage`, it redirects them to `<PAGES_URL>/docs/mypage/`, rather than just redirecting to add the slash at the end.

The larger issue here is that this causes an issue with compatibility as we are upgrading a site from v1. I have a page linked at `https://mydocs.myorg.com/docs/mypage.html`, but now i need to send them to `https://mydocs.myorg.com/docs/mypage/`. I also need to do this preserving any anchor tags that are present in the request.

Should I just expect to need to do this redirection in my nginx proxy? Is there something I can update in my docusaurus config to allow the pages to be built or linked differently?

**Comment by jmcshane** [comment]
This creates a problem for migration. V1 can respond on `docs/mypage` and `docs/mypage.html`, V2 can respond on `docs/mypage/` and `docs/mypage/index.html`. This seems like something that could be set as a build option

**Comment by jmcshane** [comment]
Ok, so I've got this solved in nginx. I needed to add the following two proxy redirect blocks next to the proxy_pass:

```
proxy_redirect http://github.myorg.com/pages/org/repo/ https://$http_host/;
proxy_redirect https://github.myorg.com/pages/org/repo/ https://$http_host/;
```

The reason for the first one is that based on some admin configurations (which I don't have access to view) GH pages redirects will respond with http rather than https. Using the redirect structure like this will allow both v1 and v2 sites to continue to work with the same proxy configuration, thus allowing a seamless migration.
